qos/cometbft/state.go

```diff
@@ -0,0 +1,94 @@
+package cometbft
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/pokt-network/poktroll/pkg/polylog"
+
+	"github.com/buildwithgrove/path/protocol"
+)
+
+// ServiceState keeps the expected current state of the CometBFT blockchain based on the endpoints' responses to
+// different requests.
+type ServiceState struct {
```

Comment:
Great! I find the consistency across QoS implementations really helpful in understanding the code (and to guide future refactors).